### PR TITLE
Improve read and write functions

### LINKED ISSUE #4: Improve read and write to the pyMBE dataframe
Currently, there are a few issues with reading and writting to the pyMBE dataframe:

- We do not directly provide a writing function `pmb.write_pmb_df(filename)` and instead expect the user to use the native function from pandas `pmb_df.to_csv(filename)`. This creates an asymmetry in the library and makes the writting of the pyMBE_df less intuitive

- The function `pmb.read_pmb_df(filename)` currently reads the pyMBE df but it does not ensure that the quantities stored on it have the same data type as in the original pmb_dataframe. For example, list variables, such as `sequence` and `residue_list` become string-like variables when reading the pmb_df from file and need to be parsed. This becomes a source of many issues when one reads the pmb_df and tries to use the rest of pyMBE functions, for example `calculate_HH`. We need to extend `pmb.read_pmb_df(filename)` to transform all variables to its original type

Issue migrated from our original remote repository on Gitlab, see here for further discusion on the topic [here](https://gitlab.com/blancoapa/pyMBE/-/issues/30)

### PR DESCRIPTION
**Pull request by paobtorres**
fixes #4

### CODE REVIEW

**Review by pm-blanco — CHANGES_REQUESTED** [review]
@paobtorres thank you for taking care of this. Could you please update the main branch of your forked repository and then merge it to this branch and fix the merge conflicts?

I also checked the CI of the last commit on your branch and it is failing with your current implementation:
```bash
python3 testsuite/peptide_tests.py
fv-az568-256:pid5411: PSM3 can't open nic unit: 1 (err=23)
fv-az568-256:pid5411: PSM3 can't open nic unit: 1 (err=23)
fv-az568-256:pid5411: PSM3 can't open nic unit: 1 (err=23)
fv-az568-256:pid5411: PSM3 can't open nic unit: 1 (err=23)
Traceback (most recent call last):
  File "/home/runner/work/pyMBE/pyMBE/samples/Beyer2024/peptide.py", line 143, in <module>
    c_salt_calculated = pmb.create_added_salt(espresso_system=espresso_system,
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/pyMBE/pyMBE/pyMBE.py", line 654, in create_added_salt
    self.create_particle(espresso_system=espresso_system, name=cation_name, number_of_particles=N_cation)
  File "/home/runner/work/pyMBE/pyMBE/pyMBE.py", line 819, in create_particle
    self.copy_df_entry(name=name,column_name='particle_id',number_of_copies=number_of_particles)
  File "/home/runner/work/pyMBE/pyMBE/pyMBE.py", line 607, in copy_df_entry
    df_by_name_repeated = self.pd.concat ([df_by_name]*(number_of_copies-1), ignore_index=True)
                                          ~~~~~~~~~~~~^^~~~~~~~~~~~~~~~~~~~
TypeError: can't multiply sequence by non-int of type 'numpy.float64'
Traceback (most recent call last):
Running tests for KKKKKDDDDD
  File "/home/runner/work/pyMBE/pyMBE/testsuite/peptide_tests.py", line 62, in <module>
    run_peptide_test(script_path=script_path,
  File "/home/runner/work/pyMBE/pyMBE/testsuite/peptide_tests.py", line 30, in run_peptide_test
    subprocess.check_output(run_command)
  File "/cvmfs/software.eessi.io/versions/2023.06/software/linux/x86_64/amd/zen3/software/Python/3.11.3-GCCcore-12.3.0/lib/python3.11/subprocess.py", line 466, in check_output
pH = 3
python3 /home/runner/work/pyMBE/pyMBE/samples/Beyer2024/peptide.py --sequence KKKKKDDDDD --pH 3 --mode test --no_verbose --output /tmp/tmpiloz45iy
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/cvmfs/software.eessi.io/versions/2023.06/software/linux/x86_64/amd/zen3/software/Python/3.11.3-GCCcore-12.3.0/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['python3', '/home/runner/work/pyMBE/pyMBE/samples/Beyer2024/peptide.py', '--sequence', 'KKKKKDDDDD', '--pH', '3', '--mode', 'test', '--no_verbose', '--output', '/tmp/tmpiloz45iy']' returned non-zero exit status 1.
make: *** [Makefile:12: tests] Error 1
```
could you please fix this?

**Comment by paobtorres** [comment]
@pm-blanco Yes, I'm currently working on it .

**Comment by pm-blanco** [comment]
@paobtorres ok, otherwise your implementation looks reasonable to me.

**Comment by paobtorres** [comment]
@pm-blanco Finally solved all the merge problems. Now is ready for review

**Comment by paobtorres** [comment]
@pm-blanco Ready for review! I adapted functions and scripts to consider parameter_of_the_potential a dict not a str.

File: pyMBE.py [hunk 1]
```diff
@@ -101,9 +101,10 @@ def add_bond_in_df(self, particle_id1, particle_id2, use_default_bond=False):
         used_bond_df = self.df.loc[self.df['particle_id2'].notnull()]
         #without this drop the program crashes when dropping duplicates because the 'bond' column is a dict
         used_bond_df = used_bond_df.drop([('bond_object','')],axis =1 )
-        if len(used_bond_df.index) > 1: 
-            used_bond_df = used_bond_df.drop_duplicates(keep='first')
+        # if len(used_bond_df.index) > 1: 
+        #     used_bond_df = used_bond_df.drop_duplicates(keep='first')
```
Comment (pm-blanco) [inline pyMBE.py:105]: can you please remove this comments from the code?

**Pull request merged**